opossum-tool/OpossumUI
# Fix console warnings caused by unnecessary rerenders

### Summary of changes

- memoize selectors to prevent rerenders using [proxy-memoize](https://github.com/dai-shi/proxy-memoize)
- simplify some selectors

### Context and reason for change

Selectors returning mutable data types such as arrays and objects should be memoized in react-redux. Otherwise they cause unnecessary rerenders. Read more here: https://react-redux.js.org/api/hooks#equality-comparisons-and-updates

I also tried using `lodash.memoize` as memoizer, or using `lodash.isEqual`/`react-redux.shallowEqual` as second argument to `useSelector`. Both approaches resulted in more rerender cycles than using `proxy-memoize`.

Closes #1259.

### How can the changes be tested

Observe the console while using the app in development mode (`yarn start`) and compare with main. You'll see that the warnings have disappeared.
